pkg/cmd/cli/cmd/idle.go

```diff
@@ -0,0 +1,557 @@
+package cmd
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	utilerrors "github.com/openshift/origin/pkg/util/errors"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	osclient "github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	deployclient "github.com/openshift/origin/pkg/deploy/client/clientset_generated/internalclientset/typed/core/unversioned"
+	unidlingapi "github.com/openshift/origin/pkg/unidling/api"
+	utilunidling "github.com/openshift/origin/pkg/unidling/util"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	clientset "k8s.io/kubernetes/pkg/client/unversioned/adapters/internalclientset"
+	"k8s.io/kubernetes/pkg/controller"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/strategicpatch"
+)
+
+const (
+	idleLong = `
+Idle scalable resources.
+
+This command idles the provides list of resources by scaling scalable resources (e.g. RCs and
+DCs) down to zero replicas, and then marking associated services so that that the scalables
+will be "woken up" when traffic occurs on those services.
+
+Only DCs and RCs with associated services will be idled -- services that only point to pods,
+as well as RCs and DCs with no services, will not be idled.`
+
+	idleExample = `  # Idle the scalable controllers associated with some services listed in to-idle.txt
+  $ %[1]s idle -f to-idle.txt`
+)
+
+// NewCmdStatus implements the OpenShift cli status command
+func NewCmdIdle(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	var inputFile string
+	dryRun := false
+	loadByService := false
+
+	cmd := &cobra.Command{
+		Use:     "idle -f FILENAME",
+		Short:   "Idle scalable resources",
+		Long:    idleLong,
+		Example: fmt.Sprintf(idleExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(RunIdle(f, out, cmd, inputFile, loadByService, dryRun))
+		},
+	}
+
+	cmd.Flags().BoolVar(&dryRun, "dry-run", false, "If true, only print the annotations that would be written, without annotating or idling the relevant objects")
+	cmd.Flags().StringVarP(&inputFile, "filename", "f", "", "file containing list of services whose scalables will be idled")
```

Comment:
Don't use "scalables".

This command needs to support the normal resource builder label selectors and name inputs, in addition to `-f -`.  So at minimum `--all`, `--all-namespaces`, `-l`, and resource type name (use the form that defaults to services, so someone can do `oc idle foo` as well as `oc idle svc/foo`.